test/junit/scala/reflect/FieldAccessTest.scala

```diff
@@ -1,14 +1,15 @@
 package scala.reflect
 
 import org.junit.Assert._
-import org.junit.Ignore
 import org.junit.Test
 
 class FieldAccessTest {
 
   class TestClass {
     private val x = 123
-    def fn = () => x
+    locally {
+      () => x
+    }
```

Comment:
restores x reference to ctor body, but didn't check what triggered the original bug. Good example of fragile sample code. Maybe an enclosing definition could be annotated with a checksum, to ensure the snapshot of code is never altered.